qiskit/circuit/quantumcircuit.py

```diff
@@ -1283,11 +1284,21 @@ def append(
             appender = self._append
             requester = self._resolve_classical_resource
         instructions = InstructionSet(resource_requester=requester)
-        for qarg, carg in operation.broadcast_arguments(expanded_qargs, expanded_cargs):
-            self._check_dups(qarg)
-            instruction = CircuitInstruction(operation, qarg, carg)
-            appender(instruction)
-            instructions.add(instruction)
+        if isinstance(operation, Instruction):
+            for qarg, carg in operation.broadcast_arguments(expanded_qargs, expanded_cargs):
+                self._check_dups(qarg)
+                instruction = CircuitInstruction(operation, qarg, carg)
+                appender(instruction)
+                instructions.add(instruction)
+        else:
+            # For Operations that are non-Instructions, we use the Instruction's default method
+            for qarg, carg in Instruction.broadcast_arguments(
+                operation, expanded_qargs, expanded_cargs
+            ):
+                self._check_dups(qarg)
+                instruction = CircuitInstruction(operation, qarg, carg)
+                appender(instruction)
+                instructions.add(instruction)
```

Comment:
Per @jakelishman's earlier comments do we want to implicitly use the instruction class's broadcast method here? I'm fine with this, but I feel like I'm missing a piece in my mental model of how we transition from this to a new model for broadcasting args without causing a potential compatibility issue down the road assuming we release this as is. I'm not sure what a good alternative is, since I haven't dug into the broadcasting piece in some time. I'm also probably missing something.

Is maybe doing something like:

```suggestion
        if hasattr(operation, "broadcast_arguments"):
            for qarg, carg in operation.broadcast_arguments(expanded_qargs, expanded_cargs):
                self._check_dups(qarg)
                instruction = CircuitInstruction(operation, qarg, carg)
                appender(instruction)
                instructions.add(instruction)
        else:
            # For Operations that are non-Instructions, we use the Instruction's default method
            for qarg, carg in Instruction.broadcast_arguments(
                operation, expanded_qargs, expanded_cargs
            ):
                self._check_dups(qarg)
                instruction = CircuitInstruction(operation, qarg, carg)
                appender(instruction)
                instructions.add(instruction)
```

a better choice? That way if an operation defined it's own broadcasting rules it'd work. But I might be thinking too much about this.

Reply:
Thanks! If I understand correctly, the decision about this during the last qiskit-dev-synthesis meeting was to leave the code as it is for now (in this stricter form), but possibly switch to what you suggest in the future.